tools/autograd/load_derivatives.py

```diff
@@ -294,10 +294,10 @@ def saved_variables(formula, args):
             'suffix': '_numel',
             'type': 'int64_t',
         }),
-        # replace to_arg_sizes(self, 2) with self_argsizes_2
-        (r'to_arg_sizes\({}, (\w+)\)', {
-            'suffix': lambda m: '_sizes_{}'.format(*m.groups()),
-            'type': 'IntList',
+        # replace to_args_sizes(self) with self_args_sizes
+        (r'to_args_sizes\({}\)', {
+            'suffix': '_args_sizes',
+            'type': 'std::vector<std::vector<int64_t>>',
```

Comment:
why can't this be std::vector<IntList> (if IntList is valid,. we should be able to hold collections of them)

Reply:
I tried using `std::vector<IntList>` instead of `std::vector<std::vector<int64_t>>` and the values inside changes between the forward and backward passes (I'm not sure why)... In addition the ArrayRef source code https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/ArrayRef.h#L23-L33 comes with a warning to not use it for storage.

Reply:
The question still remains though, why is it okay to use IntList normally?  Is it transformed into a vector somewhere?

Reply:
Yes, IntList is transformed to a std::vector<int64_t> by the code generation. I'm still trying to figure out where that happens

Reply:
here? https://github.com/pytorch/pytorch/blob/acc409396bda54e25dd0cccbd526821c52645662/tools/autograd/gen_autograd_functions.py#L139-L140.  Looks like you can just add another case here.

Reply:
I looked into it a bit more: IntList is transformed to a std::vector<int64_t> by the code generation, and one can pass in a std::vector<int64_t> into a function that takes IntList.

We don't have an analogous thing for std::vector<std::vector<int64_t>> (Unless each inner std::vector<int64_t> has the same size, but this isn't an assumption that we can make here).

The alternative to what I have now is using std::vector<IntList>. There are a few things that would need to happen to support this change:
1) Make the codegen translate `std::vector<IntList>` to `std::vector<std::vector<int64_t>>`
2) The cat_tensors_backward function will still take `std::vector<std::vector<int64_t>>`, unless we also want to change its argument to `std::vector<IntList>` for consistency. Then there needs to be an implicit conversion from `std::vector<std::vector<int64_t>>` to `std::vector<IntList>` which I'm not sure how to do.

Because of this, I think it's best to keep the usage of `std::vector<std::vector<int64_t>>` here to avoid the extra complexity -- what do you think?

Reply:
Sounds good to me, I don't think you can do the the implicit conversion.